internal/app/singularity/registry.go

```diff
@@ -134,6 +141,15 @@ func (l *Library) copyFromCache(hash, name, to string) error {
 		return errNotInCache
 	}
 
+	// Remove the 'to' image if exists, (before copying from cache).
+	err = os.Remove(to)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("unable to remote old image to overide: %s", err)
+	}
+
+	// Remove the not-fully copied image if interrupted.
+	go interruptCleanup(to)
```

Comment:
maybe remove this one from here and `pullAndVerify` and keep it in a single place?